File: JavaLab/src/main/java/ch2/section2/sub1/Quiz1.java
package ch2.section2.sub1;

public class Quiz1 {
    public static void main(String[] args) {
        int season = 11;

        if(2<season && season<6){
            System.out.println("봄입니다.");
        }else if(5<season && season<9){
            System.out.println("여름입니다.");
        }else if(8<season && season<12){
            System.out.println("가을입니다.");
        } else if (season==1 | season == 12 | season ==2) {
            System.out.println("겨울입니다");
        }
    }
}

File: KotlinLab/KotlinLab/src/main/kotlin/ch2/section7_nullsafety/q1/Quiz1.kt
package org.example.ch2.sec7_null_safety.q1

class User {
    var name: String? = null
    var address: Address? = null
}
class Address {
    var state: String = ""
    var city: String = ""
}

class Quiz1 {
    fun printUserInfo(user: User?): Unit {
        user?.let {
            try {
                println("${user.name} : ${user.address!!.state} ${user.address!!.city}")
            } catch (e: NullPointerException) {
                throw NullPointerException("주소 없음")
            }
        } ?: run {
            println("유저 정보가 없습니다.")
        }
    }
}

fun main() {
    val obj = Quiz1()
    obj.printUserInfo(null)

    val user = User()
    user.name = "홍길동"

    val address = Address()
    address.city = "성남시"
    address.state = "경기도"

    user.address = address
    obj.printUserInfo(user)
}

